src/app/services/storage.service.ts

```diff
@@ -240,4 +240,16 @@ export class StorageService {
 
     return path.indexOf('/') < 0;
   }
+
+  // Takes a string with fuzzy units like 20Mib, 20k, 20Gb, returns number in bytes
+  convertDataMeasurementStrings(str: string) {
+    if(!str) { return 0 }; // if empty, return the default 0, which is unlimited
+    if(str[0] && !str[0].match(/[0-9]/)) { return NaN };
+    const letters = str.replace( /[^a-zA-Z]/g, '');
+    if (letters[0] && !letters[0].toLowerCase().match(/b|k|m|g|t/i)) { return NaN };
+    const powersOf1024 = {b: 1, k: 1024, m: 1024**2, g: 1024**3, t: 1024**4};
+    const unit = str.toLowerCase().match(/[b|k|m|g|t]/i) || 'b';
+    const value = str.match(/^\D*(\d+)/) ? str.match(/^\D*(\d+)/)[1] : '0';
```

Comment:
Hm.  Can't really skip over non-digit characters (\D), because those could be unit numbers that should not be allowed before a number, like "g 1".  Better to only match what is allowed.  Any number of whitespace characters followed by one or more digits is what I'd use.This works (although only trivially tested):
```
const value = (num = str.match(/^\s*(\d+)/)) ? num[1] : '0';
```

Reply:
I cannot get this to accept the leading spaces.

Reply:
As written, this validates anything that doesn't match by replacing it with zero. If you change '0' at the end to NaN, it throws the error if you enter a space, but will clear the err message and accept it if you eventually follow with a number.